--- php-4.0.6.orig/ext/mbstring/mbfilter.c
+++ php-4.0.6/ext/mbstring/mbfilter.c
@@ -95,18 +95,18 @@
 #define	mbfl_pfree		free
 
 /* unicode table */
-extern const unsigned short iso8859_2_ucs_table[];
-extern const unsigned short iso8859_3_ucs_table[];
-extern const unsigned short iso8859_4_ucs_table[];
-extern const unsigned short iso8859_5_ucs_table[];
-extern const unsigned short iso8859_6_ucs_table[];
-extern const unsigned short iso8859_7_ucs_table[];
-extern const unsigned short iso8859_8_ucs_table[];
-extern const unsigned short iso8859_9_ucs_table[];
-extern const unsigned short iso8859_10_ucs_table[];
-extern const unsigned short iso8859_13_ucs_table[];
-extern const unsigned short iso8859_14_ucs_table[];
-extern const unsigned short iso8859_15_ucs_table[];
+static const unsigned short iso8859_2_ucs_table[];
+static const unsigned short iso8859_3_ucs_table[];
+static const unsigned short iso8859_4_ucs_table[];
+static const unsigned short iso8859_5_ucs_table[];
+static const unsigned short iso8859_6_ucs_table[];
+static const unsigned short iso8859_7_ucs_table[];
+static const unsigned short iso8859_8_ucs_table[];
+static const unsigned short iso8859_9_ucs_table[];
+static const unsigned short iso8859_10_ucs_table[];
+static const unsigned short iso8859_13_ucs_table[];
+static const unsigned short iso8859_14_ucs_table[];
+static const unsigned short iso8859_15_ucs_table[];
 
 /* charactor property table */
 #define MBFL_CHP_CTL		0x01
@@ -116,7 +116,7 @@
 #define MBFL_CHP_MMHQENC	0x10	/* must Q-encoding in MIME Header encoded-word */
 #define MBFL_CHP_MSPECIAL	0x20	/* RFC822 Special characters */
 
-extern const unsigned char mbfl_charprop_table[];
+static const unsigned char mbfl_charprop_table[];
 
 
 /* language structure */
--- php-4.0.6.orig/ext/mbstring/mbfilter_ja.c
+++ php-4.0.6/ext/mbstring/mbfilter_ja.c
@@ -83,35 +83,35 @@
 #include "mbfilter_ja.h"
 
 /* cp932 table */
-extern const unsigned short cp932ext3_eucjp_table[];
-extern const int cp932ext3_eucjp_table_size;
+static const unsigned short cp932ext3_eucjp_table[];
+static const int cp932ext3_eucjp_table_size;
 
 /* unicode table */
-extern const unsigned short jisx0208_ucs_table[];
-extern const unsigned short jisx0212_ucs_table[];
-extern const unsigned short cp932ext1_ucs_table[];
-extern const unsigned short cp932ext2_ucs_table[];
-extern const unsigned short cp932ext3_ucs_table[];
-extern const unsigned short ucs_a1_jis_table[];
-extern const unsigned short ucs_a2_jis_table[];
-extern const unsigned short ucs_i_jis_table[];
-extern const unsigned short ucs_r_jis_table[];
-extern const int jisx0208_ucs_table_size;
-extern const int jisx0212_ucs_table_size;
-extern const int cp932ext1_ucs_table_min;
-extern const int cp932ext1_ucs_table_max;
-extern const int cp932ext2_ucs_table_min;
-extern const int cp932ext2_ucs_table_max;
-extern const int cp932ext3_ucs_table_min;
-extern const int cp932ext3_ucs_table_max;
-extern const int ucs_a1_jis_table_min;
-extern const int ucs_a1_jis_table_max;
-extern const int ucs_a2_jis_table_min;
-extern const int ucs_a2_jis_table_max;
-extern const int ucs_i_jis_table_min;
-extern const int ucs_i_jis_table_max;
-extern const int ucs_r_jis_table_min;
-extern const int ucs_r_jis_table_max;
+static const unsigned short jisx0208_ucs_table[];
+static const unsigned short jisx0212_ucs_table[];
+static const unsigned short cp932ext1_ucs_table[];
+static const unsigned short cp932ext2_ucs_table[];
+static const unsigned short cp932ext3_ucs_table[];
+static const unsigned short ucs_a1_jis_table[];
+static const unsigned short ucs_a2_jis_table[];
+static const unsigned short ucs_i_jis_table[];
+static const unsigned short ucs_r_jis_table[];
+static const int jisx0208_ucs_table_size;
+static const int jisx0212_ucs_table_size;
+static const int cp932ext1_ucs_table_min;
+static const int cp932ext1_ucs_table_max;
+static const int cp932ext2_ucs_table_min;
+static const int cp932ext2_ucs_table_max;
+static const int cp932ext3_ucs_table_min;
+static const int cp932ext3_ucs_table_max;
+static const int ucs_a1_jis_table_min;
+static const int ucs_a1_jis_table_max;
+static const int ucs_a2_jis_table_min;
+static const int ucs_a2_jis_table_max;
+static const int ucs_i_jis_table_min;
+static const int ucs_i_jis_table_max;
+static const int ucs_r_jis_table_min;
+static const int ucs_r_jis_table_max;
 
 #define CK(statement)	do { if ((statement) < 0) return (-1); } while (0)
 
