--- php-4.0.0.orig/sapi/apache/mod_php4.c
+++ php-4.0.0/sapi/apache/mod_php4.c
@@ -331,7 +331,7 @@
 }
 
 
-static sapi_module_struct sapi_module = {
+static sapi_module_struct apache_sapi_module = {
 	"Apache",						/* name */
 									
 	php_apache_startup,				/* startup */
@@ -587,8 +587,8 @@
 #ifdef ZTS
 		tsrm_startup(1, 1, 0);
 #endif
-		sapi_startup(&sapi_module);
-		php_apache_startup(&sapi_module);
+		sapi_startup(&apache_sapi_module);
+		php_apache_startup(&apache_sapi_module);
 	}
 	per_dir_entry.type = mode;
 
@@ -672,7 +672,7 @@
 static void apache_php_module_shutdown_wrapper(void)
 {
 	apache_php_initialized = 0;
-	sapi_module.shutdown(&sapi_module);
+	apache_sapi_module.shutdown(&apache_sapi_module);
 #ifdef ZTS
 	tsrm_shutdown();
 #endif
@@ -687,8 +687,8 @@
 #ifdef ZTS
 		tsrm_startup(1, 1, 0);
 #endif
-		sapi_startup(&sapi_module);
-		php_apache_startup(&sapi_module);
+		sapi_startup(&apache_sapi_module);
+		php_apache_startup(&apache_sapi_module);
 	}
 #if MODULE_MAGIC_NUMBER >= 19980527
 	{
--- php-4.0.0.orig/sapi/cgi/cgi_main.c
+++ php-4.0.0/sapi/cgi/cgi_main.c
@@ -179,7 +179,7 @@
 
 
 
-static sapi_module_struct sapi_module = {
+static sapi_module_struct cgi_sapi_module = {
 	"CGI",							/* name */
 									
 	php_module_startup,				/* startup */
@@ -380,7 +380,7 @@
 	tsrm_startup(1,1,0);
 #endif
 
-	sapi_startup(&sapi_module);
+	sapi_startup(&cgi_sapi_module);
 
 #ifdef PHP_WIN32
 	_fmode = _O_BINARY;			/*sets default for file streams to binary */
@@ -426,7 +426,7 @@
 #endif							/* FORCE_CGI_REDIRECT */
 	}
 
-	if (php_module_startup(&sapi_module)==FAILURE) {
+	if (php_module_startup(&cgi_sapi_module)==FAILURE) {
 		return FAILURE;
 	}
 #ifdef ZTS
--- php-4.0.0.orig/sapi/cgi/php_getopt.h
+++ php-4.0.0/sapi/cgi/php_getopt.h
@@ -3,7 +3,5 @@
 
 extern char *ap_php_optarg;
 extern int ap_php_optind;
-extern int ap_php_opterr;
-extern int ap_php_optopt;
 
 int ap_php_getopt(int argc, char* const *argv, const char *optstr);
