--- php-4.3.0.orig/ext/mysql/php_mysql.c
+++ php-4.3.0/ext/mysql/php_mysql.c
@@ -105,6 +105,14 @@
 	(mysql_field_count(mysql)>0)
 #endif
 
+#if !defined(MYSQL_PORT) && defined(MARIADB_PORT)
+#define MYSQL_PORT		MARIADB_PORT
+#endif
+
+#if !defined(MYSQL_UNIX_ADDR) && defined(MARIADB_UNIX_ADDR)
+#define MYSQL_UNIX_ADDR		MARIADB_UNIX_ADDR
+#endif
+
 typedef struct _php_mysql_conn {
 	MYSQL conn;
 	int active_result_id;
Only in php-4.3.0: ext/mysql/php_mysql.c.orig
