--- php-4.3.2.orig/sapi/apache/config.m4
+++ php-4.3.2/sapi/apache/config.m4
@@ -85,7 +85,7 @@
                       \$(mkinstalldirs) '$APXS_SYSCONFDIR' && \
                        $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
                              -S SYSCONFDIR='$APXS_SYSCONFDIR' \
-                             -i -a -n php4 $SAPI_SHARED"
+                             -i -n php4 $SAPI_SHARED"
     fi
   fi
 
